android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java

```diff
@@ -108,6 +108,11 @@ public void updateEmail(String email) {
 
     @ReactMethod
     public void getEmail(Promise promise) {
+        String email = RNIterableInternal.getEmail();
+        if (email == null){
+            promise.reject("", "email is null");
```

Comment:
Is this the expected behavior for a promise? I think `null` is a normal return value if the user hasn't logged in yet. I'm not sure about the semantics of a `Promise` but I thought it should only return a 'rejected' state if some error occurred?

Reply:
We have to come to a consensus for that. This way RN developers dont have to rely on null checks for getEmail to proceed with their operations. And its like guarantee that something is always there if its a success.

Reply:
@tapashmajumder What do you think?

Reply:
Had a chat with Tapash. Null email case will go to promise.resolve.